tests/test_api.py

```diff
@@ -563,3 +554,40 @@ def test_history():
     assert isinstance(estimator.history_, dict)
     assert all(isinstance(k, str) for k in estimator.history_.keys())
     assert all(isinstance(v, list) for v in estimator.history_.values())
+
+
+def test_compile_model_from_params():
+    """Tests that if build_fn returns an un-compiled model,
+    the __init__ parameters will be used to compile it.
+    """
+    # Load data
+    data = load_boston()
+    X, y = data.data[:100], data.target[:100]
+
+    # build_fn that does not compile
+    def build_fn_uncompiled():
+        model = Sequential()
+        model.add(keras.layers.Dense(X.shape[1], input_shape=(X.shape[1],)))
+        model.add(keras.layers.Activation("relu"))
+        model.add(keras.layers.Dense(1))
+        model.add(keras.layers.Activation("linear"))
+        return model
+
+    # and one that does
+    def build_fn_compiled(myloss):
```

Comment:
Style nit to make this a little cleaner:

``` python
def build_fn(compile=True, compile_kwargs=None):
    ...
    if compile:
        model.compile(**compile_kwargs)
    return model
```

Reply:
Didn't exactly work as a drop in replacement, but I took the idea and molded it in.